### PR TITLE
[Model Enhancement]: Colour Detection using DL

### PR DESCRIPTION
**Pull request by Arihant-Bhandari**
# Pull Request for DL-Simplified 💡

## Issue Title : [Model Enhancement]: Colour Detection using DL

- **Info about the related issue (Aim of the project)** : Enhancing  custom CNN model for Colour Detection
- **Name:** Arihant Bhandari
- **GitHub ID:** Arihant-Bhandari https://github.com/Arihant-Bhandari
- **Email ID:** arihant.bhandari.2004@gmail.com
- **Idenitfy yourself:** **GSSoC 2024 Contributor** 

Closes: #529 [Model Enhancement]: Colour Detection using DL
### Describe the add-ons or changes you've made 📃

i have enchanced the previous model by adding the new image loading function from keras.utils, Attention mechanism, using ELU instead of ReLU due to its better functionality, made the problem into a multi-class classification model with softmax implementation, used the optimizer Nadam with lr = 0.01, displayed the first 32 kernels based on the images, saved and included the best model as part of the PR.  

## Type of change ☑️

What sort of change have you made:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Code style update (formatting, local variables)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## How Has This Been Tested? ⚙️

The model was tested on Accuracxy, Precision, F1-score, Recall and confusion matrtix:

Accuracy: 0.94

Precision: 0.9565476190476192

Recall: 0.9514285714285714

F1 Score: 0.95005439005439

Confusion Matrix:
[[6 0 0 0 1 0 0 0 0 0]
 [0 4 0 0 0 0 0 0 0 0]
 [0 0 7 0 0 0 0 0 0 0]
 [0 0 0 2 0 0 0 0 0 0]
 [0 0 0 0 6 0 0 0 0 0]
 [0 0 0 0 0 4 0 0 0 1]
 [0 0 1 0 0 0 6 0 0 0]
 [0 0 0 0 0 0 0 5 0 0]
 [0 0 0 0 0 0 0 0 2 0]
 [0 0 0 0 0 0 0 0 0 5]]

## Checklist: ☑️

- [x] My code follows the guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly wherever it was hard to understand.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added things that prove my fix is effective or that my feature works.
- [ ] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

**Review by abhisheks008 — CHANGES_REQUESTED** [review]
Hi @Arihant-Bhandari your model looks good to me. As you are enhancing the existing project, add your cells inside the existing `.ipynb` file and document that properly inside the `.ipynb` file. Also update the existing `README` as per your model accuracy score.

**Comment by Arihant-Bhandari** [comment]
hi @abhisheks008 i wanted to know if i should completely change the README file, since someone worked on it yesterday, and it is signed off with their name, should i keep mine in tandem or delete theirs and add mine, i wanted to know before finalizing on it.

**Comment by abhisheks008** [comment]
> hi @abhisheks008 i wanted to know if i should completely change the README file, since someone worked on it yesterday, and it is signed off with their name, should i keep mine in tandem or delete theirs and add mine, i wanted to know before finalizing on it.

Put it like this,
README Enhanced by, <Contributor Name>
Model Enhanced by, <Your Name>

We should give credit to the contributor who enhanced and corrected the README earlier.
